### DISCUSSION TITLE
Display ST7789 320*240 ripples when power is applied

### DISCUSSION THREAD

**Discussion by brightproject** [Bodmer/TFT_eSPI#2778] [General]
I am sure that `99%` of the problem is not related to the library.
The defect appears only when the display is turned on for the first time, incl. power supply.
![photo1691234596](https://github.com/Bodmer/TFT_eSPI/assets/1788098/d81a6b8e-3eb2-403a-a9ed-a95c0370ba5a)
Then everything works without ripples.
I tried to install a shunt capacitor to power the display, but there is no change.
The display is powered by `3.3` volts from the `esp32` LDO regulator.
I feed the `esp32` itself from `USB.`
I also tried to power it from the power bank - but nothing changes.
The distortion pattern itself is always the same - as if it was the `ST7789` driver that controls it.
Upper half, left 9/10 parts horizontal dots, 1/10 part vertical dots.
Bottom half, left 9/10 parts horizontal stripes, 1/10 half - a few dots vertically.
Wrote in support of the manufacturer of the display driver - `Sitronix Technology Corporation`, but so far have not received a response.
I tried to play with the display drivers in the settings, but there is no effect either.
I tried to initialize the display with the command `init();` and `begin();`
It would be desirable that when the device was turned on, the display would simply be black, and then display the image.
And here it turns out ripples, distortions - not a presentation of the device in general!
P.S.
As I thought - the problem of malnutrition.
Start sequences followed in the library

> https://github.com/Bodmer/TFT_eSPI/blob/1986bd242391243d9b29e2f9e50eae1b9c705825/TFT_Drivers/ST7789_Init.h#L136

It turns out that the `LDO,` which is installed on the esp32 board, does not allow maintaining a voltage level of 3.3 volts at the start of the microcontroller - `esp32` sin with a large current at startup, which sometimes reaches `0.9 mA`.
When the display was powered from a separate 3.3 volt power supply, the launch became without ripples and problems.
Quality nutrition is very important, especially in the era of modern microcontrollers.

**Comment by riacob** [comment]
You could
- Add a delay before initializing the display - this will circumvent the natually high boot current of the ESP32
- Add a capacitor to the 3v3 line, compensating for high boot current consumption
